frontend/packages/dev-console/src/components/monitoring/alerts/SilenceAlert.tsx

```diff
@@ -20,14 +27,35 @@ const SilenceUntil = ({ rule }) => {
 };
 
 const SilenceAlert: React.FC<SilenceAlertProps> = ({ rule }) => {
-  const [isChecked, setIsChecked] = React.useState(rule.state !== RuleStates.Silenced);
-  const [ruleState] = React.useState(rule);
-  React.useEffect(() => setIsChecked(ruleState.state !== RuleStates.Silenced), [ruleState]);
+  const [isChecked, setIsChecked] = React.useState(true);
+  const rules = useSelector((state: RootState) => state.UI.getIn(['monitoring', 'devRules']));
+  const dispatch = useDispatch();
+  React.useEffect(() => {
+    if (rule.state === RuleStates.Silenced) {
+      setIsChecked(false);
+    }
+  }, [rule]);
 
   const handleChange = (checked: boolean) => {
     if (checked) {
       _.each(rule.silencedBy, (silence) => {
-        coFetchJSON.delete(`${alertManagerBaseURL}/api/v2/silence/${silence.id}`);
+        coFetchJSON
+          .delete(`${alertManagerBaseURL}/api/v2/silence/${silence.id}`)
+          .then(() => {
```

Comment:
In the admin Console, after deleting a silence, we call `refreshNotificationPollers`, which immediately fetches the latest silences from Alertmanager and then triggers the evaluation of which alerts are silenced. I think that would achieve the same thing you are doing here in a simpler way.